pyiron_atomistics/vasp/base.py

```diff
@@ -1856,6 +1747,127 @@ def list_potentials(self):
+        if scheme == "GC":
+            if mesh is None:
+                mesh = [int(val) for val in self.input.kpoints[3].split()]
+            self.input.kpoints.set_kpoints_file(
+                size_of_mesh=mesh, shift=center_shift, method="Gamma centered"
+            )
+        if scheme == "GP":
+            self.input.kpoints.set_kpoints_file(
+                size_of_mesh=[1, 1, 1], method="Gamma Point"
+            )
+        if scheme == "Line":
+            if n_path is None and self.input.kpoints._n_path is None:
+                raise ValueError("n_path has to be defined")
+            high_symmetry_points = self.structure.get_high_symmetry_points()
+            if high_symmetry_points is None:
+                raise ValueError("high_symmetry_points has to be defined")
+
+            if path_name is None and self.input.kpoints._path_name is None:
+                raise ValueError("path_name has to be defined")
+            if path_name not in self.structure.get_high_symmetry_path().keys():
+                raise ValueError("path_name is not a valid key of high_symmetry_path")
+
+            if path_name is not None:
+                self.input.kpoints._path_name = path_name
+            if n_path is not None:
+                self.input.kpoints._n_path = n_path
+
+            self.input.kpoints.set_kpoints_file(
+                method="Line",
+                n_path=self.input.kpoints._n_path,
+                path=self._get_path_for_kpoints(self.input.kpoints._path_name),
+            )
+        if scheme == "Manual":
+            if manual_kpoints is None:
+                raise ValueError(
+                    "For the manual mode, the kpoints list should be specified"
+                )
+            else:
+                if weights is not None:
+                    if not (len(manual_kpoints) == len(weights)):
+                        raise AssertionError()
+                self.input.kpoints.set_value(line=1, val=str(len(manual_kpoints)))
+                if reciprocal:
+                    self.input.kpoints.set_value(line=2, val="Reciprocal")
+                else:
+                    self.input.kpoints.set_value(line=2, val="Cartesian")
+                for i, kpt in enumerate(manual_kpoints):
+                    if weights is not None:
+                        wt = weights[i]
+                    else:
+                        wt = 1.0
+                    self.input.kpoints.set_value(
+                        line=3 + i,
+                        val=" ".join([str(kpt[0]), str(kpt[1]), str(kpt[2]), str(wt)]),
+                    )
+
+    def _reinit_potential_setter(self, structure):
+        if structure is not None:
+            self._potential.to_dict().update(
+                {
+                    el: None
+                    for el in set(structure.get_chemical_symbols())
+                    if el not in self._potential.to_dict().keys()
+                }
+            )
```

Comment:
If I understand correctly this relies on `self._potential.to_dict()` returning its internal `dict` directly and that modifications on the return value are reflected in the object.  That can be dangerous, I think.

```suggestion
            for el in structure.get_species_symbols():
                if el not in self._potential.to_dict():
                    self._potential[el] = None
```

Thought it would require that `VaspPotentialSetter.__getitem__/.__getattr__` don't raise errors on new elements anymore.

Reply:
> If I understand correctly this relies on `self._potential.to_dict()` returning its internal `dict` directly and that modifications on the return value are reflected in the object. That can be dangerous, I think.

I agree that it is dangerous, that is why I added the test. When a copy of the dictionary is returned the test breaks - I just checked this again. From my perspective this keeps the functionality of the `VaspPotentialSetter` class as defined previously, like raising errors when users mistype element names or try to set an element which does not exist in the structure and it does not directly access private variables which causes IDEs like Pycharm to complain. If anybody is willing to rewrite the handling of pseudo potentials in VASP I am open for suggestions, still that would be a separate pull request. 

So from my perspective this is the minimal change required to fix the bug Raynol reported and I would like to merge it as is, to prevent other users from struggling with the same issue.

Reply:
The only other issue I thought of is:  What happens when the newly set structure has less elements than the `VaspPotentialSetter` had previously defined.  When POTCAR is written are only those elements written that are actually in the structure or all that are in `VaspPotentialSetter`?  I'm not familar enough with the code to immediately tell, but that may cause issues if we accidentally write POTCARs of additional elements.  If that works, go ahead.

Reply:
We loop over the element types in the structure when we gather the paths of the potentials:
https://github.com/pyiron/pyiron_atomistics/blob/main/pyiron_atomistics/vasp/potential.py#L429
So having elements in the potential list which do not appear in the structure is not harmful. The more risky part is changing the element type of the structure, like:
```
job_vasp.structure[:] = "Al"
```
This was an issue before and that is still an issue now, but I also do not know a clever way to change it at the moment.

Reply:
Fair enough then. I suppose your case could be avoided if we added the potentials as tags or something in the structure itself, but that's a different change entirely.